rust-src/wallet_library/src/credential.rs

```diff
@@ -82,11 +83,37 @@ pub fn create_unsigned_credential_with_seed_v1_aux(
         wallet.get_blinding_randomness(identity_provider_index, identity_index)?;
     let encoded_blinding_randomness = base16_encode_string(&blinding_randomness);
 
+    let mut attribute_randomness = BTreeMap::new();
+    for attribute_name in ATTRIBUTE_NAMES.iter() {
+        let tag = AttributeTag::from_str(attribute_name).unwrap();
+        let randomness: PedersenRandomness<ArCurve> = wallet.get_attribute_commitment_randomness(
+            identity_provider_index,
+            identity_index,
+            input.common.cred_number.into(),
+            tag,
+        )?;
+        attribute_randomness.insert(tag, randomness);
```

Comment:
I don't understand why we are generating all randomness here. In the `UnsignedCredentialInput` we have the tags in the identity object. Should we not be generating randomness only for those (or rather only for those that are not revealed)?

Reply:
I copied what we had in the JS SDK that already did this. Looking at the code further down I can see that we indeed only use the randomness for the non-revealed attributes. I will fix it.